packages/caravan-psbt/src/psbtv2/psbtv2.ts

```diff
@@ -787,6 +787,70 @@ export class PsbtV2 extends PsbtV2Maps {
     }
   }
 
+  /**
+   * Sets the sequence number for a specific input in the transaction.
+   *
+   * This private helper method is crucial for implementing RBF and other
+   * sequence-based transaction features. It writes the provided sequence
+   * number as a 32-bit little-endian unsigned integer and stores it in the
+   * appropriate input's map using the PSBT_IN_SEQUENCE key.
+   *
+   * The sequence number has multiple uses in Bitcoin transactions:
+   * 1. Signaling RBF (values < 0xfffffffe)
+   * 2. Enabling nLockTime (values < 0xffffffff)
+   * 3. Relative timelock with BIP68 (if bit 31 is not set)
+   *
+   * Care should be taken when setting sequence numbers to ensure the desired
+   * transaction properties are correctly signaled.
+   */
+  private setInputSequence(inputIndex: number, sequence: number) {
```

Comment:
Correct, this would need to be a public method in order to provide the Updater role with the ability to add a sequence value to an input.

As for the locktime, well, I see now that I added some TODOs in the `addInput` method for handling locktimes. So, you can currently add a sequence without a locktime on the input (is this valid). At the very least, the global fallback locktime will be used. Since the locktime is outside the scope of your change, maybe just ensure this checks `isReadyForUpdater` for now and leave the locktime checking for another time.

Reply:
This is also all still a little vague for me as well, but based on [this](https://bitcoin.stackexchange.com/questions/120256/what-is-the-relation-between-nlocktime-and-nsequence) I think that leaving locktime for another time is. My read is that checking for locktime as well is if you're trying to enforce timelocks, but if you're just using it for signaling then setting the sequence is fine. 

This should be easily verifiable though. update a tx with this (can be in regtest) and see if the node accepts it.

Reply:
changes done ... (here)[https://github.com/caravan-bitcoin/caravan/pull/114/commits/07d5a8431fa4cf6a5e79e83e04649b34023c2a7e#diff-cefa8ce7b34d9637fc1bfd4f54edc413efd6e74068ebf1e97e4ab8cad063ee2fR827]

added tests [too](https://github.com/caravan-bitcoin/caravan/pull/114/commits/07d5a8431fa4cf6a5e79e83e04649b34023c2a7e#diff-7b81e397987df31b2946b614d1abcbda706e2bced67fa519706fa4f51688e540R1597) :)